### ISSUE TITLE
[🐞] Bug with Rollup Packaging in TS Files

### ISSUE THREAD

**Issue by jdgamble555** [BuilderIO/qwik#3487]
### Which component is affected?

Qwik Rollup / Vite plugin

### Describe the bug

My Simple Firebase app does not seem to work when I import `signInWithPopup` or the basic `getAuth()`.  After trying to downgrade Firebase and Vite, I finally realized this is a problem with something that was done in one of the latest Qwik versions.

### Reproduction

https://github.com/jdgamble555/qwik-firebase-todo

### Steps to reproduce

Install the app in this branch `https://github.com/jdgamble555/qwik-firebase-todo/tree/f333f884b88e428f769d603b1d16ddb8307440b6`.  The difference being that `getAuth()` and `signInWithPopup` are imported inside a `ts` file, instead of a `tsx` file.

While the app technically works now (on the current commit), it is a hack.  I need to be able to share common imports in my `ts` file, not just in a `tsx` file. 

### System Info

```shell
You can see the errors running `npm run build`, or you can deploy to vercel to see the errors.

https://qwik-firebase-todo.vercel.app/ - working now finally
```

### Additional Information

I believe something was changed in the last 2-3 updates that effects the way Rollup does tree shaking.  My files in the `ts` are not added, yet they work fine when added to `tsx` files.

**Comment by jdgamble555** [comment]
I was able to create **temporary** fix by dynamically importing BOTH `getAuth()` and `signInWithPopup`. This doesn't seem to effect other Firebase imports for some reason.

https://github.com/jdgamble555/qwik-firebase-todo/blob/master/src/lib/firebase.ts

J

**Comment by jdgamble555** [comment]
So, I believe the fix for the above [linked issue](https://github.com/BuilderIO/qwik/issues/3297) fixed this problem.  It is also is clear now with the error message that you CANNOT import `getAuth()` into the server, as we have seen with other server side frameworks.

The solution, as I linked to [my repo](https://github.com/jdgamble555/qwik-firebase-todo/blob/master/src/lib/firebase.ts) above, is to either:

1.) Dynamically import the firebase functions when uses (greater performance)
2.) Check for browser when importing the functions

I used **1)** in my repo, and I recommend that version.

J

This issue can be closed. 😃
